cranelift/frontend/src/ssa.rs

```diff
@@ -360,26 +366,26 @@ impl SSABuilder {
     pub fn remove_block_predecessor(&mut self, block: Block, inst: Inst) -> Block {
         debug_assert!(!self.is_sealed(block));
         let pred = self.ssa_blocks[block].remove_predecessor(inst);
-        self.update_predecessor_cycle(block, pred, false);
+        self.update_predecessor_cycle(block, pred, Action::Remove);
         pred
     }
 
     /// Call this when `pred` is not in the predecessors of `block`: before adding, or after
     /// removing.
-    fn update_predecessor_cycle(&mut self, block: Block, mut pred: Block, add: bool) {
+    fn update_predecessor_cycle(&mut self, block: Block, mut pred: Block, action: Action) {
         let block_ref = &self.ssa_blocks[block];
 
         // Not counting `pred`, how many predecessors does `block` have? And will it have exactly
         // one after the caller is done changing things?
         let will_have_one_pred = match block_ref.predecessors.as_slice() {
             // None: `pred` might reach `block` in a cycle. There'll be one predecessor afterward
             // if we're adding `pred`.
-            [] => add,
+            [] => matches!(action, Action::Add),
```

Comment:
a tiny thing but if you add `#[derive(PartialEq, Eq)]` (or maybe just `PartialEq`? I'm never clear on what needs what) to the enum, then this can be `action == Action::Add`, which I personally like a bit more for zero-arg enums (C-style enums). Totally up to you though.

Reply:
Oh, yeah, that is definitely nicer. Thanks for the suggestion!

I think `PartialEq` is technically enough but since the derived implementation is actually total, I think it's confusing to leave off `Eq`, so I'm going with both.